### PR TITLE
Sitediff API class and refactorings.

### PR DESCRIPTION
**Pull request by cleaver**
- new SiteDiff::Api class
- move business logic to the API class
- make config use more consistent
- general code cleanups

### CODE REVIEW

File: lib/sitediff/config/creator.rb [hunk 1]
```diff
@@ -14,10 +14,10 @@ class Config
     class Creator
       ##
       # Creates a Creator object.
-      def initialize(debug, *urls)
```
Comment (cleaver) [inline lib/sitediff/config/creator.rb:17]: Order is important, so I think this is more clear.

File: lib/sitediff/config/creator.rb [hunk 2]
```diff
@@ -31,9 +31,11 @@ def roots
 
       ##
       # Build a config structure, return it.
-      def create(options, &block)
```
Comment (cleaver) [inline lib/sitediff/config/creator.rb:34]: This callback was not used.

File: lib/sitediff/config.rb [hunk 3]
```diff
@@ -194,9 +194,10 @@ def self.merge(first, second)
     # Merges 2 iterable objects deeply.
     def self.merge_deep(first, second)
       first.merge(second) do |_key, val1, val2|
-        if val1.is_a? Hash
+        case val1.class
+        when Hash
           self.class.merge_deep(val1, val2 || {})
-        elsif val1.is_a? Array
```
Comment (cleaver) [inline lib/sitediff/config.rb:199]: Rubocop violation

File: sitediff.gemspec [hunk 4]
```diff
@@ -2,8 +2,8 @@
 
 Gem::Specification.new do |s|
   s.name        = 'sitediff'
-  s.version     = '1.0.0'
-  s.required_ruby_version = '>= 2.4'
+  s.version     = '1.1.1'
+  s.required_ruby_version = '>= 2.6'
```
Comment (gardon) [inline sitediff.gemspec:6]: We need to update docs if we will require 2.6 now

**Pull request merged**